guides/plugins/plugins/content/cms/add-cms-element-via-admin-sdk.md

```diff
@@ -0,0 +1,352 @@
+
+// Only execute extensionSDK commands when
+// it is inside a iFrame (only needed for plugins)
+if (location.isIframe()) {
+    if (location.is(location.MAIN_HIDDEN)) {
+        // Execute the base commands
+        import('./base/mainCommands');
+    } else {
+        // Render different views
+        import('./viewRenderer');
+    }
+}
+```
+
+This is main file, which is executed first and functions as our entry point.
+
+We start with `if(location.isIframe())` to make sure we only load content used inside iFrames. The reason for this is that the SDK can be used in apps and plugins, so this check makes sure that your code is executed in the right place.
+
+Next we need `if(location.is(location.MAIN_HIDDEN))` to **load the main commands**, which are defined in the `mainCommands.ts` file. This will only be used to load logic, but not templates into the Administration.
+
+Lastly the `else` case will be responsible for specific loading of views via `viewRenderer.ts`. This is where our view templates will be loaded.
+
+### Loading all required templates
+
+Now we need to create the `viewRenderer.ts` file, which will include the 3 mandatory files needed for a CMS element. We prefix them with `swag-dailymotion`, consisting of a vendor prefix and the component name, to make sure no other developer will accidentally choose the same name as we did:
+
+- `swag-dailymotion-config.ts`, which will handle the content of the CMS element configuration
+- `swag-dailymotion-element.ts`, which represents the actual target element in the CMS
+- `swag-dailymotion-preview.ts`, which is responsible for the preview, when selecting the CMS element in its selection screen
+
+Let's see how the component loading via `viewRenderer.ts` looks like:
+
+```ts
+import Vue from 'vue';
+import { location } from '@shopware-ag/admin-extension-sdk';
+
+// watch for height changes
+location.startAutoResizer();
+
+// start app views
+const app = new Vue({
+    el: '#app',
+    data() {
+        return { location };
+    },
+    components: {
+        'SwagDailymotionElement':
+            () => import('./views/swag-dailymotion/swag-dailymotion-element'),
+        'SwagDailymotionConfig':
+            () => import('./views/swag-dailymotion/swag-dailymotion-config'),
+        'SwagDailymotionPreview':
+            () => import('./views/swag-dailymotion/swag-dailymotion-preview'),
+    },
+    template: `
+        <SwagDailymotionElement
+            v-if="location.is('swag-dailymotion-element')"
+        ></SwagDailymotionElement>
+        <SwagDailymotionConfig
+            v-else-if="location.is('swag-dailymotion-config')"
+        ></SwagDailymotionConfig>
+        <SwagDailymotionPreview
+            v-else-if="location.is('swag-dailymotion-preview')"
+        ></SwagDailymotionPreview>
+    `,
+});
+```
+
+Really straight forward, isn't it? As you probably know from vanilla VueJS's Options API, we just need to load, register and use our VueJS component to make them work.
```

Comment:
Vue.js instead of VueJS.

Reply:
What's "Vanilla VueJS"? Are you mentioning Composition API somewhere in this article or any other mechanisms? Otherwise I think it's an unnecessary detail :)

Reply:
I meant non-shopware vue, but sure, this can be removed :D